vlm/src/main/scala/geotrellis/contrib/vlm/TileToLayout.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.contrib.vlm
+
+import geotrellis.layer.{LayoutDefinition, SpaceTimeKey, SpatialKey}
+import geotrellis.raster.ResampleMethod
+import geotrellis.raster.resample.NearestNeighbor
+
+/**
+  * Applies the given [[LayoutDefinition]] to the source data producing a [[LayoutTileSource]].
+  * In order to fit to the given layout, the source data is resampled to match the Extent
+  * and CellSize of the layout.
+  */
+trait TileToLayout[K] extends Serializable {
```

Comment:
`TileToLayout` has to be a type class to choose the instance in compile time depending on the Key type.